### PR TITLE
Automate Rails Code Library Release

### PR DESCRIPTION
**Pull request by smusali**
The context is completely similar to [logdna/ruby#13](https://github.com/logdna/ruby/pull/13)

### CODE REVIEW

File: logdna-rails.gemspec [hunk 1]
```diff
@@ -1,24 +1,22 @@
...
-  spec.homepage      = 'https://github.com/logdna/rails'
-  spec.license       = 'MIT'
+  spec.summary       = "LogDNA Rails Logger"
+  spec.homepage      = "https://github.com/logdna/rails"
+  spec.license       = "MIT"
 
-  spec.files         = ["lib/logdna-rails.rb"]
-  spec.bindir        = 'exe'
-  spec.require_paths = ['lib']
+  spec.files         = `git ls-files -z`.split("\x0").select { |f| f.match(/(.md|.rb|.gemspec|Gemfile|LICENSE)$/) }
```
Comment (vilyapilya) [inline logdna-rails.gemspec]: Revert this to the way it was. There is no need to make it depended on git.

**Review by vilyapilya — COMMENTED** [review]
changes requested

File: logdna-rails.gemspec (hunk 1 above)
Comment (smusali) [reply logdna-rails.gemspec]: @vilyapilya, that's how it's in the `Best Practices` - even in [`logdna/ruby`](github.com/logdna/ruby), it was with `git ls-files -z` - that's the recommended way of specifying files inside `gemspec` + some ownership-related files were missing initially

Comment (vilyapilya) [reply logdna-rails.gemspec]: I have a pr that changes it in Ruby. This is not a recommended way. That way it adds a redundant prerequisite of having git for creating the gem, but there is no need to have git there. Besides, there is no need to make so completed regex, it is a small file

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+              echo "  TAG_VERSION: ${CIRCLE_TAG}"
+              echo "  PKG_VERSION: v${PKG_VERSION}"
+              exit 1
+            fi
+      - run: gem build logdna-rails.gemspec
+      - persist_to_workspace:
+          root: .
+          paths:
+            - ./logdna-rails-*.gem
+  app2pub:
```
Comment (mindjiver) [inline .circleci/config.yml]: What does this step actually do? It attaches the workspace and saves the already existing file there? Or is it somehow used for the manual approval step?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      - run: rubocop -c .rubocop.yml -a
+  build:
+    docker:
+      - image: circleci/ruby:2.7.0
+    steps:
+      - checkout
+      - run:
+          name: Check Tagged Push
+          command: |
+            PKG_VERSION=$(cat lib/version.rb | grep VERSION | cut -d'"' -f2)
```
Comment (mindjiver) [inline .circleci/config.yml]:
```suggestion
            PKG_VERSION=$(grep VERSION lib/version.rb | cut -d'"' -f2)
```

File: .circleci/config.yml (hunk 2 above)
Comment (smusali) [reply .circleci/config.yml]: Yes - previously, I tried on `agent` changing type of the functional job to `approval` but it didn't work - basically, there should be a separate job for approval stage

File: logdna-rails.gemspec (hunk 1 above)
Comment (smusali) [reply logdna-rails.gemspec]: As it says [here](https://bundler.io/v1.16/guides/creating_gem.html#getting-started), it's using `git ls-files` when it builds and releases the gem; so, it's not creating any extra redundant prerequisite on the client-side and the place where it's getting built and released already has `git`

And [here](https://bundler.io/v1.16/guides/creating_gem.html#getting-started), it explains why it's not just `.rb` files to include

The reason of having `git` here is to pick up only committed files and prevent failures, as said.

**Review by mindjiver — APPROVED** [review]
From the CircleCI side of things it's looking good. My Ruby is a bit rusty but it all looked alright to me.

File: logdna-rails.gemspec (hunk 1 above)
Comment (vilyapilya) [reply logdna-rails.gemspec]: Gemspec is used when the gem is installed. And this creates a requirement for a user to have git initiated in the folder where they're trying to install git. The only file we need to include is the logdna-rails.rb. It is sufficient just to write this file explicitly.

Comment (vilyapilya) [reply logdna-rails.gemspec]: It is good to use git in the circleci scripts, git is the integral part of it. But git should not be in the prereq for building dependencies.

**Pull request merged**